src/components/ExperienceDetail/ReportModal/useReportModal.js

```diff
@@ -0,0 +1,27 @@
+import { useState, useCallback } from 'react';
+
+export function useReportModal() {
+  const [modalState, setModal] = useState({
+    isModalOpen: false,
+    modalType: '',
+    modalPayload: {},
+  });
+
+  const [closableOnClickOutside, setModalClosableOnClickOutside] = useState(
+    true,
+  );
+
+  const handleIsModalOpen = useCallback(
+    (isModalOpen, modalType = '', modalPayload = {}) => {
+      setModal({ isModalOpen, modalType, modalPayload });
+    },
+    [],
+  );
+
+  return {
+    modalState,
+    handleIsModalOpen,
```

Comment:
`set` 感覺比較像 setter (主動要產生事件)，`handler` 感覺是 event handler (監聽事件）
```suggestion
    setModalOpen,
```